File: C++_program/Problem_on_Numbers/SpyNumber.cpp
//////////////////////////////////////////////////////////////////////////////////////////////
//
//required header file
//
//////////////////////////////////////////////////////////////////////////////////////////////

#include<iostream>                   // for taking input from user
using namespace std ;

//////////////////////////////////////////////////////////////////////////////////////////////
//
//  Class Name  :   Logic
//  Function Name : SpyNumber()
//  Description :   check number is Spy number or not
//  Input :         Integer
//  output :        void
//  Author :        Rekha Shankarlal Kumawat
//  Date :          22/03/2025
//
//////////////////////////////////////////////////////////////////////////////////////////////

class Logic
{
    public:
    int iNo ;

    Logic(int iNo)
    {
        this->iNo = iNo;
    }
    bool SpyNumber()
    {

        int iSum = 0 ;
        int iprod = 0 ;
        int iDigit = 0; 
        int iTemp = 0;
        
        iTemp = this->iNo ;

        iprod = 1 ;
        while(this->iNo != 0)
        {
            iDigit = this->iNo % 10 ;
            iSum = iSum + iDigit;
            iprod = iprod*iDigit;

            this->iNo = this->iNo /10 ;
        }
        return iSum == iprod;
        
    }
};
int main()
{
    int iValue = 0;
    bool bRet = false ;

    cout<<"Enter the Number :- \n";
    cin>>iValue;
    
    Logic lobj(iValue);
    bRet =  lobj.SpyNumber();

    if(bRet == true)
    {
        cout<<iValue<<" is a Spy Number\n";
    }
    else
    {
        cout<<iValue<<" is not a Spy Number\n";
        
    }

    return 0 ;
}

File: Python_Programming/Problems_on_Numbers/SpyNumber.py
'''
    Description : program  to check the number is Spy number or not 
                EX :- sum of digit = product of digit
                
    Date :- 21/03/2026
    
    Author : Rekha Shankarlal Kumawat
                
'''
def SpyNumber(iNo):


    iSum = 0 
    iprod = 0 
    iDigit = 0 
    iTemp = 0
    
    iTemp = iNo 

    iprod = 1 
    while(iNo != 0):
    
        iDigit = iNo % 10 
        iSum = iSum + iDigit
        iprod = iprod*iDigit

        iNo = iNo //10 
    
    return iSum == iprod
    
def main():
    
    iValue = 0
    bRet = False 

    print("Enter the Number :- ")
    iValue = int(input())
    
    bRet =  SpyNumber(iValue)

    if(bRet == True):
        print(iValue ,"is a Spy Number")
    else:
        print(iValue ,"is not a Spy Number")

if __name__ == "__main__":
    main()